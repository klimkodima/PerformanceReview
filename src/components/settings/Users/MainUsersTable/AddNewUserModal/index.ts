export { default as AddNewUserModal } from './AddNewUserModalContainer';
