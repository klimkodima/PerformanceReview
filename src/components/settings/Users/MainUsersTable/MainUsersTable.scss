.users-table {
  &__container {
    max-height: calc(100vh - 450px);
  }
  td {
    font-weight: 600;
  }
  td:nth-of-type(1) {
    font-weight: 600;
  }
  td:nth-of-type(1),
  td:nth-of-type(2) {
    text-align: left;
  }
  .disabled-row {
    td {
      color: #b3b3b3;
    }
  }
}

#add-user-button {
  margin: 20px 0;
}
