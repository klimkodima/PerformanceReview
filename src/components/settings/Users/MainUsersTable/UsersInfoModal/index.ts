export { default as UserInfoModal } from './UserInfoModalContainer';
