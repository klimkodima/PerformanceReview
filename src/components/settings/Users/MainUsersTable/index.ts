export { default as MainUsersTable } from './MainUsersTable';
