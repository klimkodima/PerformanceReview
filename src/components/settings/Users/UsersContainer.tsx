import { ReactElement } from 'react';

import { MainUsersTable } from './MainUsersTable';
import { PendingUsersTable } from './PendingUsersTable';

const UsersContainer = (): ReactElement => (
  <>
    <MainUsersTable />
    <PendingUsersTable />
  </>
);

export default UsersContainer;
