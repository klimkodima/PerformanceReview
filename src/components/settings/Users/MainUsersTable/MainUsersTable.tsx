import { ReactElement, useState, useEffect } from 'react';

import { t } from 'i18next';
import { useSelector, useDispatch } from 'react-redux';
import {
  TableContainer,
  Paper,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  TableSortLabel,
  Button
} from '@mui/material';

import { createFormattedLevel } from 'src/utils';
import { selectSettingsPermission } from 'src/store/App';
import {
  fetchUsers,
  fetchUsersRefresh,
  resetCurrentUserInfo,
  selectUsers,
  setCurrentUserInfo,
  setIsUsersTableOpened
} from 'src/store/Users';
import { useDebounce } from 'src/components/hooks';
import { selectCurrentUser } from 'src/store/Auth';
import { ROLE_NAME, SETTINGS_PERMISSION } from 'src/constants';
import { descendingSort } from 'src/components/common/CommonTable/constants';
import { Search } from 'src/components/common';
import { UserInfoModal } from './UsersInfoModal';
import { AddNewUserModal } from './AddNewUserModal';
import {
  ENABLED_KEY,
  HeaderCellsUserType,
  HEADER_CELLS_USERS,
  Levels,
  LEVEL_KEY,
  LEVEL_LABEL,
  Roles,
  ROLE_LABEL,
  ROLE_NAME_KEY,
  TEAM_LABEL,
  USERS_ROW_CELLS
} from '../constants';

import { UserInfoType } from 'src/store/Users/types';

import './MainUsersTable.scss';

const MainUsersTable = (): ReactElement => {
  const settingsPermission = useSelector(selectSettingsPermission);
  const isEdit = settingsPermission === SETTINGS_PERMISSION.WRITE;
  const dispatch = useDispatch();
  const usersData = useSelector(selectUsers);
  const { roleName } = useSelector(selectCurrentUser);

  const isAdmin = roleName === ROLE_NAME.ADMIN;

  useEffect(() => {
    dispatch(fetchUsers());
    dispatch(setIsUsersTableOpened(true));
    dispatch(fetchUsersRefresh());

    return () => {
      dispatch(setIsUsersTableOpened(false));
    };
  }, []);

  const [orderBy, setOrderBy] = useState<string>(t('users_table.team_label'));
  const [isUserInfoOpen, setIsUserInfoOpen] = useState<boolean>(false);
  const [isAddNewUserOpen, setIsAddNewUserOpen] = useState<boolean>(false);

  const handleRequestSort = (property: string): void => {
    setOrderBy(property);
  };

  const mainUsers = usersData.filter((user) => !user.pending);
  mainUsers.sort((a, b) => a.fullName.localeCompare(b.fullName));

  const sortData = (mainUsers: UserInfoType[], orderBy: string) => {
    if (orderBy === TEAM_LABEL) {
      const usersWithoutTeam = mainUsers.filter((user) => !user.teamId);
      const usersWithTeam = mainUsers.filter((user) => user.teamId);
      usersWithTeam.sort((a, b) => (a.teamId as number) - (b.teamId as number));

      return [...usersWithTeam, ...usersWithoutTeam].sort((a, b) =>
        a.enabled === b.enabled ? 0 : a.enabled ? -1 : 1
      );
    }

    if (orderBy === ROLE_LABEL) {
      return mainUsers
        .sort((a, b) => Roles[a.roleName] - Roles[b.roleName])
        .sort((a, b) => (a.enabled === b.enabled ? 0 : a.enabled ? -1 : 1));
    }

    if (orderBy === LEVEL_LABEL) {
      const usersWithoutLevel = mainUsers.filter((user) => !user.level);
      const usersWithLevel = mainUsers.filter((user) => user.level);
      usersWithLevel.sort(
        (a, b) => Levels[a.level as string] - Levels[b.level as string]
      );

      return [...usersWithLevel, ...usersWithoutLevel].sort((a, b) =>
        a.enabled === b.enabled ? 0 : a.enabled ? -1 : 1
      );
    }
  };

  const sortedRows = sortData(mainUsers, orderBy);

  const getCellValue = (cell: string, value: string): string => {
    if (cell === ENABLED_KEY) return value ? 'Active' : 'Inactive';
    if (cell === ROLE_NAME_KEY || cell === LEVEL_KEY) {
      return value ? createFormattedLevel(value) : '-';
    }
    return value ? value : '-';
  };

  const [searchValue, setSearchValue] = useState<string>('');
  const [searchUsers, setSearchUsers] = useState(sortedRows);
  const debouncedSearchTerm = useDebounce(searchValue, 1000);

  const handleSearchUsers = (): void => {
    const searchUsers = sortedRows?.filter((user: UserInfoType): boolean =>
      user.fullName.toLowerCase().includes(searchValue.toLowerCase())
    );

    setSearchUsers(searchUsers);
  };

  useEffect(() => {
    if (debouncedSearchTerm) {
      handleSearchUsers();
    } else {
      setSearchUsers(sortedRows);
    }
  }, [debouncedSearchTerm]);

  const handleSearchValue = (value: string): void => {
    setSearchValue(value);
  };

  const usersList = !searchValue ? sortedRows : searchUsers;

  const handleUserRowDoubleClick = (user: UserInfoType) => (): void => {
    dispatch(setCurrentUserInfo(user));
    setIsUserInfoOpen(true);
  };

  const handleCloseModalClick = (): void => {
    dispatch(resetCurrentUserInfo());
    setIsUserInfoOpen(false);
  };

  const handleAddNewUserButtonClick = (): void => {
    setIsAddNewUserOpen(true);
  };

  const handleCloseAddNewUserModalClick = (): void => {
    setIsAddNewUserOpen(false);
  };

  useEffect(() => {
    if (isUserInfoOpen || isAddNewUserOpen)
      document.body.classList.add('scroll');
    return () => {
      document.body.classList.remove('scroll');
    };
  }, [isUserInfoOpen, isAddNewUserOpen]);

  return (
    <div>
      <Search
        handleSearchValue={handleSearchValue}
        searchValue={searchValue}
        placeholder={t('search.placeholder')}
      />
      <div className='settings-table'>
        <div className='settings-table__wrapper users-table'>
          <TableContainer component={Paper} className='users-table__container'>
            <Table aria-label='table'>
              <TableHead>
                <TableRow>
                  {HEADER_CELLS_USERS.map(
                    ({ name, isShowIcon }: HeaderCellsUserType) => (
                      <TableCell key={name}>
                        {isShowIcon && (
                          <TableSortLabel
                            active={orderBy === name}
                            direction={descendingSort}
                            onClick={() => handleRequestSort(name)}
                          >
                            {name}
                          </TableSortLabel>
                        )}
                        {!isShowIcon && name}
                      </TableCell>
                    )
                  )}
                </TableRow>
              </TableHead>
              <TableBody>
                {usersList?.map((user: UserInfoType) => {
                  const inactiveStyle = user.enabled ? '' : 'disabled-row';
                  return (
                    <TableRow
                      key={user.id}
                      className={inactiveStyle}
                      onDoubleClick={handleUserRowDoubleClick(user)}
                    >
                      {USERS_ROW_CELLS.map((cell: string) => (
                        <TableCell key={cell}>
                          {getCellValue(cell, user[cell] as string)}
                        </TableCell>
                      ))}
                    </TableRow>
                  );
                })}
              </TableBody>
            </Table>
          </TableContainer>
        </div>
      </div>
      {isAdmin && (
        <Button
          id='add-user-button'
          onClick={handleAddNewUserButtonClick}
          variant='outlined'
        >
          {t('add_new_user_modal.save_button')}
        </Button>
      )}
      {isUserInfoOpen && (
        <UserInfoModal
          onCloseModalClick={handleCloseModalClick}
          isEditable={isEdit}
        />
      )}
      {isAddNewUserOpen && (
        <AddNewUserModal onCloseModalClick={handleCloseAddNewUserModalClick} />
      )}
    </div>
  );
};

export default MainUsersTable;
