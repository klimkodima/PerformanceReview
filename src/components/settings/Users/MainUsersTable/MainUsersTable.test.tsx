import { I18nextProvider } from 'react-i18next';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux';

import store from 'src/store';
import i18n from 'src/i18n';
import { t } from 'i18next';
import { setUsers } from 'src/store/Users';
import { setAvailableWidgets } from 'src/store/App';
import mockUsersData from 'src/store/Users/mockUsers.json';
import MainUsersTable from './MainUsersTable';
import UsersContainer from '../UsersContainer';

beforeEach(() => {
  store.dispatch(setUsers(mockUsersData));
  store.dispatch(
    setAvailableWidgets({
      settingsPermission: 'WRITE',
      availableWidgets: []
    })
  );
});

describe('Main Users table tests', () => {
  test('Main Users table snapshot', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <MainUsersTable />
          </I18nextProvider>
        </Provider>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('Users Container has the elements', () => {
    const { getAllByRole } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <UsersContainer />
        </I18nextProvider>
      </Provider>
    );

    expect(getAllByRole('table')).toHaveLength(2);
    expect(getAllByRole('heading', { level: 2 })).toHaveLength(1);
  });

  test('Main Users table has the elements', () => {
    const { getByRole, getAllByRole, getAllByTestId } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <MainUsersTable />
        </I18nextProvider>
      </Provider>
    );

    expect(getByRole('table')).toBeInTheDocument();
    expect(getAllByRole('columnheader')).toHaveLength(6);
    expect(getAllByRole('row')).toHaveLength(10);
    expect(getAllByRole('cell')).toHaveLength(54);
    const arrowIcons = getAllByTestId('ArrowDownwardIcon');
    expect(arrowIcons).toHaveLength(3);
    userEvent.click(arrowIcons[0]);
    userEvent.click(arrowIcons[1]);
    userEvent.click(arrowIcons[2]);
  });

  test('Main Users table has search element', () => {
    jest.useFakeTimers();
    const { getByRole, getByTestId } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <MainUsersTable />
        </I18nextProvider>
      </Provider>
    );

    const searchInput = getByRole('textbox');
    expect(searchInput).toBeInTheDocument();
    expect(searchInput).toHaveAttribute('placeholder', t('search.placeholder'));
    userEvent.type(searchInput, 'Hello');
    jest.advanceTimersByTime(2000);
    expect(searchInput).toHaveValue('Hello');
    const resetBtn = getByTestId('reset-button');
    expect(resetBtn).toBeInTheDocument();
    userEvent.click(resetBtn);
    expect(searchInput).toHaveValue('');
    jest.clearAllTimers();
  });
});
