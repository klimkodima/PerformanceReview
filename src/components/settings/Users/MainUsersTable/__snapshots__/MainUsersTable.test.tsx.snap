// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Users table tests Main Users table snapshot 1`] = `
<div>
  <div
    className="search"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium search__icon css-i4bv87-MuiSvgIcon-root"
      data-testid="SearchIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
    <div
      className="MuiInputBase-root MuiInputBase-colorPrimary css-1le28he-MuiInputBase-root"
      onClick={[Function]}
    >
      <input
        aria-label="search"
        className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Enter User Name"
        type="text"
        value=""
      />
    </div>
    <button
      aria-label="reset"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium search__hidden css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      data-testid="reset-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="reset"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
  <div
    className="settings-table"
  >
    <div
      className="settings-table__wrapper users-table"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root users-table__container css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          aria-label="table"
          className="MuiTable-root css-rqglhn-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                User Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Email
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Status
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Role
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc css-3l415a-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Level
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc css-3l415a-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <span
                  className="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Team
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc css-3l415a-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Anna Ivanova
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                a.ivalova@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Active
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Auditor
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Middle
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                X-Rays
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Tatiana Klimchenya
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                t.klimchenya@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Active
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Team Leader
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Team Lead
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                X-Rays
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Alla Shishkin
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                a.shishkin@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Active
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Auditor
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                -
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Charlie
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Igor Egorov
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                i.egorov@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Active
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Manager
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Team Lead
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Charlie
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Alla Elenova
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                a.elenova@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Active
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Auditor
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                -
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Alpha
              </td>
            </tr>
            <tr
              className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Egor Egorov
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                e.egorov@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Active
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Admin
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Senior
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                -
              </td>
            </tr>
            <tr
              className="MuiTableRow-root disabled-row css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Anna Anova
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                a.anova@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Inactive
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Auditor
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Junior
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Charlie
              </td>
            </tr>
            <tr
              className="MuiTableRow-root disabled-row css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                A Shishkin
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                a.shishkin@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Inactive
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Auditor
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Trainee
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                -
              </td>
            </tr>
            <tr
              className="MuiTableRow-root disabled-row css-1q1u3t4-MuiTableRow-root"
              onDoubleClick={[Function]}
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Ivan Shishkin
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                i.shishkin@gpsolutions.com
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Inactive
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Auditor
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Trainee
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                -
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
